pkg/workflows/compute_test.go

```diff
@@ -0,0 +1,204 @@
+					Ref: "data-feeds-report",
+					Inputs: workflows.StepInputs{
+						Mapping: map[string]any{"observations": "$(Compute.outputs.Value)"},
+					},
+					Config: map[string]any{
+						"aggregation_config": ocr3.DataFeedsConsensusConfigAggregationConfig{
+							AllowedPartialStaleness: "false",
+							Feeds: map[string]ocr3.FeedValue{
+								anyFakeFeedID: {
+									Deviation: "0.5",
+									Heartbeat: 3600,
+								},
+							},
+						},
+						"aggregation_method": "data_feeds",
+						"encoder":            "EVM",
+						"encoder_config": ocr3.DataFeedsConsensusConfigEncoderConfig{
+							Abi: "(bytes32 FeedID, uint224 Price, uint32 Timestamp)[] Reports",
+						},
+						"report_id": "0001",
+					},
+					CapabilityType: capabilities.CapabilityTypeConsensus,
+				},
+			},
+			Targets: []workflows.StepDefinition{
+				{
+					ID: "write_ethereum-testnet-sepolia@1.0.0",
+					Inputs: workflows.StepInputs{
+						Mapping: map[string]any{"signed_report": "$(data-feeds-report.outputs)"},
+					},
+					Config: map[string]any{
+						"address":    "0xE0082363396985ae2FdcC3a9F816A586Eed88416",
+						"deltaStage": "45s",
+						"schedule":   "oneAtATime",
+					},
+					CapabilityType: capabilities.CapabilityTypeTarget,
+				},
+			},
+		}
+
+		assertWorkflowSpec(t, expectedSpec, spec)
+	})
+
+	t.Run("compute runs the function and returns the value", func(t *testing.T) {
+		workflow := createWorkflow(convertFeed)
+
+		fn := workflow.GetFn("Compute")
+		require.NotNil(t, fn)
+
+		req := capabilities.CapabilityRequest{Inputs: nsf}
+		actual := fn(struct{}{}, req)
+		require.NoError(t, actual.Err)
+
+		expected := [][]streams.Feed{
+			{
+				{
+					BenchmarkPrice:       "12.3",
+					FeedId:               anyFakeFeedID,
+					FullReport:           "report",
+					ObservationTimestamp: 123,
+					ReportContext:        "context",
+					Signatures:           []string{"sig1", "sig2"},
+				},
+			},
+		}
+
+		computed := &workflows.ComputeOutput[[][]streams.Feed]{}
+		err = actual.Value.UnwrapTo(computed)
+		require.NoError(t, err)
+
+		assert.Equal(t, expected, computed.Value)
+	})
+
+	t.Run("compute returns errors correctly", func(t *testing.T) {
+		anyErr := errors.New("nope")
+		workflow := createWorkflow(func(_ workflows.SDK, inputFeed notstreams.Feed) ([][]streams.Feed, error) {
+			return nil, anyErr
+		})
+
+		fn := workflow.GetFn("Compute")
+		require.NotNil(t, fn)
+
+		req := capabilities.CapabilityRequest{Inputs: nsf}
+		actual := fn(struct{}{}, req)
+		require.Equal(t, anyErr, actual.Err)
+	})
+}
+
+func createWorkflow(fn func(_ workflows.SDK, inputFeed notstreams.Feed) ([][]streams.Feed, error)) *workflows.WorkflowSpecFactory {
+	workflow := workflows.NewWorkflowSpecFactory(workflows.NewWorkflowParams{
+		Owner: "owner",
+		Name:  "name",
+	})
+
+	trigger := notstreams.TriggerConfig{MaxFrequencyMs: 5000}.New(workflow)
+	computed := workflows.Compute1(workflow, "Compute", workflows.Compute1Inputs[notstreams.Feed]{
```

Comment:
Why have ComputeN as a generated name instead of having a single Compute with N as a parameter?  Won't that cause problem for the Engine? Can it build a function name in runtime?

Reply:
A parameter where? The engine won't care it just maps the inputs form what the workflow spec has as inputs.  It has no idea if

Compute { Arg0: b, Arg1: d} is two objects or one inside the compute.

Reply:
I misunderstood - thought that N references the N-th compute step in a workflow.